server/test/read.go

```diff
@@ -673,11 +567,119 @@ func ReadQueryErrorTest(t *testing.T, datastore storage.OpenFGADatastore) {
 			require.NoError(err)
 
 			test.request.StoreId = store
-			if test.request.AuthorizationModelId == "" {
-				test.request.AuthorizationModelId = test.model.Id
-			}
 			_, err = commands.NewReadQuery(datastore, tracer, logger, encoder).Execute(ctx, test.request)
-			require.EqualError(err, test.err.Error())
+			require.Error(err)
 		})
 	}
 }
+
+func ReadAllTuplesTest(t *testing.T, datastore storage.OpenFGADatastore) {
+	ctx := context.Background()
+	tracer := telemetry.NewNoopTracer()
+	logger := logger.NewNoopLogger()
+	store := ulid.Make().String()
+
+	model := &openfgapb.AuthorizationModel{
+		Id:            ulid.Make().String(),
+		SchemaVersion: typesystem.SchemaVersion1_0,
+		TypeDefinitions: []*openfgapb.TypeDefinition{
+			{
+				Type: "repo",
+			},
+		},
+	}
+
+	err := datastore.WriteAuthorizationModel(ctx, store, model)
+	require.NoError(t, err)
+
+	writes := []*openfgapb.TupleKey{
+		{
+			Object:   "repo:openfga/foo",
+			Relation: "admin",
+			User:     "github|jon.allie@openfga",
+		},
+		{
+			Object:   "repo:openfga/bar",
+			Relation: "admin",
+			User:     "github|jon.allie@openfga",
+		},
+		{
+			Object:   "repo:openfga/baz",
+			Relation: "admin",
+			User:     "github|jon.allie@openfga",
+		},
+	}
+	err = datastore.Write(ctx, store, nil, writes)
+	require.NoError(t, err)
+
+	cmd := commands.NewReadQuery(datastore, tracer, logger, encoder.NewBase64Encoder())
+
+	firstRequest := &openfgapb.ReadRequest{
+		StoreId:           store,
+		PageSize:          wrapperspb.Int32(1),
+		ContinuationToken: "",
+	}
+	firstResponse, err := cmd.Execute(ctx, firstRequest)
+	require.NoError(t, err)
+
+	require.Len(t, firstResponse.Tuples, 1)
+	require.NotEmpty(t, firstResponse.ContinuationToken)
+
+	var receivedTuples []*openfgapb.TupleKey
+	for _, tuple := range firstResponse.Tuples {
+		receivedTuples = append(receivedTuples, tuple.Key)
+	}
+
+	secondRequest := &openfgapb.ReadRequest{StoreId: store, ContinuationToken: firstResponse.ContinuationToken}
+	secondResponse, err := cmd.Execute(ctx, secondRequest)
+	require.NoError(t, err)
+
+	require.Len(t, secondResponse.Tuples, 2)
+	require.Empty(t, secondResponse.ContinuationToken)
```

Comment:
Shouldn't there be a third page? There are 3 tuples and the page_size=1 so there should be 3 Reads.

Reply:
Two requests are done. One with page_size=1 and one with page_size=2. No continuation token because that seems to be different for every endpoint 🤷

Reply:
Whoops. Sorry. I misread it as well. Three tuples are written, the first request has page_size=1, the second request the continuation token and the default page size (of 50) so gets back the remaining two tuples, and no continuation token.

Reply:
Ah, gotcha! Thanks for the clarification. That makes sense.